csharp/src/Drivers/BigQuery/BigQueryConnection.cs

```diff
@@ -0,0 +1,954 @@
+            if (pattern == null)
+                return ".*";
+
+            StringBuilder builder = new StringBuilder("(?i)^");
+            string convertedPattern = pattern.Replace("_", ".").Replace("%", ".*");
+            builder.Append(convertedPattern);
+            builder.Append("$");
+
+            return builder.ToString();
+        }
+
+        private string ToTypeName(string type)
+        {
+            int index = Math.Min(type.IndexOf("("), type.IndexOf("<"));
+            string dataType = index == -1 ? type : type.Substring(0, index);
+            return dataType;
+        }
+
+        private XdbcDataType ToXdbcDataType(string type)
+        {
+            switch (type)
+            {
+                case "INTEGER" or "INT64":
+                    return XdbcDataType.XdbcDataType_XDBC_INTEGER;
+                case "FLOAT" or "FLOAT64":
+                    return XdbcDataType.XdbcDataType_XDBC_FLOAT;
+                case "BOOL" or "BOOLEAN":
+                    return XdbcDataType.XdbcDataType_XDBC_BIT;
+                case "STRING":
+                    return XdbcDataType.XdbcDataType_XDBC_VARCHAR;
+                case "BYTES":
+                    return XdbcDataType.XdbcDataType_XDBC_BINARY;
+                case "DATETIME":
+                    return XdbcDataType.XdbcDataType_XDBC_DATETIME;
+                case "TIMESTAMP":
+                    return XdbcDataType.XdbcDataType_XDBC_TIMESTAMP;
+                case "TIME":
+                    return XdbcDataType.XdbcDataType_XDBC_TIME;
+                case "DATE":
+                    return XdbcDataType.XdbcDataType_XDBC_DATE;
+                case "RECORD" or "STRUCT":
+                    return XdbcDataType.XdbcDataType_XDBC_VARBINARY;
+                case "NUMERIC" or "DECIMAL" or "BIGNUMERIC" or "BIGDECIMAL":
+                    return XdbcDataType.XdbcDataType_XDBC_NUMERIC;
+
+                default:
+                    return XdbcDataType.XdbcDataType_XDBC_UNKNOWN_TYPE;
+            }
+        }
+
+        public override Schema GetTableSchema(string catalog, string dbSchema, string tableName)
+        {
+            string query = string.Format("SELECT * FROM `{0}`.`{1}`.INFORMATION_SCHEMA.COLUMNS WHERE table_name = '{2}'",
+                catalog, dbSchema, tableName);
+
+            BigQueryResults result = this.client.ExecuteQuery(query, parameters: null);
+
+            List<Field> fields = new List<Field>();
+
+            foreach (BigQueryRow row in result)
+            {
+                fields.Add(DescToField(row));
+            }
+
+            return new Schema(fields, null);
+        }
+
+        private Field DescToField(BigQueryRow row)
+        {
+            Dictionary<string, string> metaData = new Dictionary<string, string>();
+            metaData.Add("PRIMARY_KEY", "");
+            metaData.Add("ORDINAL_POSITION", row["ordinal_position"].ToString());
+            metaData.Add("DATA_TYPE", row["data_type"].ToString());
```

Comment:
Should these annotations be standardized for ADBC?

Reply:
I'm not aware of a particular standard for the Field metadata. It seems like it's up to the provider at this point.